### PR TITLE
Move kv store wrapper to cosmos package

### PR DESCRIPTION
**Pull request by NicolasMahe**
Dependent on https://github.com/mesg-foundation/engine/pull/1456

This PR moves the kv store wrapper to the cosmos but also adds the ability to marshal and unmarshal data using the codec.
The goal is to move in SDK the typing version of our current database without all the logic of the current database files.
The PR also updates runner sdk to use it. The goal is for all sdk to use it so we can delete completely the database package.

Check the 2 following commits:
- https://github.com/mesg-foundation/engine/pull/1475/commits/c308223c5566e1ab69e473021382717f5695d361
- https://github.com/mesg-foundation/engine/pull/1475/commits/7baedf1b0507eac3d5671b23085c382a68a18dad

----

TODO in other PRs:
- [ ] update service sdk
- [ ] update ownership sdk
- [ ] update instance sdk

### CODE REVIEW

File: database/store/cosmos.go [hunk 1]
```diff
@@ -38,7 +38,26 @@ func (s *CosmosStore) Has(key []byte) (has bool, err error) {
 	return
 }
 
-// Get retrives service from store. It returns an error if the store does not contains the key.
+// All returns every value from the store.
+func (s *CosmosStore) All() (out [][]byte, err error) {
```
Comment (krhubert) [inline database/store/cosmos.go]: This one is rather unused? why we want it here?

Comment (NicolasMahe) [reply database/store/cosmos.go]: It was a try and i guess i include the commit by mistake!

File: cosmos/db.go [hunk 2]
```diff
@@ -11,18 +11,9 @@ import (
 
 // wrapRecoverError recover error from panic and wrap it into a db error.
 func wrapRecoverError(err *error) {
-	if *err != nil {
-		return
+	if r := recover(); *err == nil && r != nil {
+		*err = fmt.Errorf("db: %s", r)
```
Comment (NicolasMahe) [inline cosmos/db.go]: i really want to wrap the error into `fmt.Errorf("db: %w", errR)`. that why the code was a bit long

**Comment by NicolasMahe** [comment]
Closing this PR in favor of https://github.com/mesg-foundation/engine/pull/1482

**Pull request closed without merging**